docs/developer/adr/0001-remove-isinstance-checks-when-setting-parameters.md

```diff
@@ -23,13 +23,38 @@ Therefore, we should remove this mechanism and replace it with a more general va
 The more general validation mechanism can be considered out of scope for Sciline, and should be implemented in the user code or using other common libraries such as `pydantic`.
 
 Finally, we can think of this mechanism as a form of runtime type checking.
-This is not the intended scope of Sciline.
+We should ask ourselves if this is the intended scope of Sciline.
+If it is, shouldn't we also check that each provider actually returns the correct type?
+
+The main problem with not checking value types when setting parameters is that it is not possible to catch such errors with `mypy`, in contrast to return values of providers, which `mypy` *can* check.
+
+Consider the following example of setting $Q$ bins for a workflow, given by a `scipp.Variable`, which would then be passed to `scipp.hist` to create a histogram:
+
+```python
+pipeline[QBins] = sc.linspace(...)
+pipeline[QBins] = 1000  # error in current implementation
+pipeline[QBins] = sc.linspace(..., unit='m')  # no error, but wrong unit
+```
+
+Checking the type catches the first error, but not the second.
+Paradoxically, setting an integer would often be a valid operation in the example, since `scipp.hist` can handle this case, whereas the wrong unit would not be valid.
+This may indicate that defining `QBins` as an alias of `scipp.Variable` is actually an anti-pattern.
```

Comment:
```suggestion
This may indicate that defining `QBins` as an alias of `scipp.Variable` is not precise enough.
```
It's a too specific case to call it a pattern.